assets/js/amp-block-validation.js

```diff
@@ -0,0 +1,185 @@
+/*jshint esversion: 6 */
+/*global _, wp:true */
+/**
+ * AMP Gutenberg integration.
+ *
+ * On editing a block, this checks that the content is AMP-compatible.
+ * And it displays a notice if it's not.
+ */
+
+/* exported ampBlockValidation */
+var ampBlockValidation = ( function( $ ) {
+	var module = {
+
+		/**
+		 * Holds data.
+		 */
+		data: {},
+
+		/**
+		 * Boot module.
+		 *
+		 * @param {Object} data Object data.
+		 * @return {void}
+		 */
+		boot: function( data ) {
+			module.data = data;
+			$( document ).ready( function() {
+				if ( 'undefined' !== typeof wp.blocks ) {
+					module.processBlocks();
+				}
+			} );
+		},
+
+		/**
+		 * Gets all of the registered blocks, and overwrites their edit() functions.
+		 *
+		 * The new edit() functions will check if the content is AMP-compliant.
+		 * If not, it will display a notice.
+		 *
+		 * @returns {void}
+		 */
+		processBlocks: function() {
+			var blocks = wp.blocks.getBlockTypes();
+			blocks.forEach( function( block ) {
+				if ( block.hasOwnProperty( 'name' ) ) {
+					module.overwriteEdit( block.name );
+				}
+			} );
+		},
+
+		/**
+		 * Overwrites the edit() function of a block.
+		 *
+		 * Outputs the original edit function, stored in OriginalBlockEdit.
+		 * This also appends a notice to the block.
+		 * It only displays if the block's content isn't valid AMP,
+		 *
+		 * @see https://riad.blog/2017/10/16/one-thousand-and-one-way-to-extend-gutenberg-today/
+		 * @param {string} blockType the type of the block, like 'core/paragraph'.
+		 * @returns {void}
+		 */
+		overwriteEdit: function( blockType ) {
+			let block = wp.blocks.unregisterBlockType( blockType );
+			let OriginalBlockEdit = block.edit;
+
+			block.edit = class AMPNotice extends wp.element.Component {
```

Comment:
ES6 classes aren't going to work in IE11.

Reply:
Thanks, @westonruter. [This commit](https://github.com/Automattic/amp-wp/pull/1019/commits/ea1c37af19209b6851b261cb115105a5b5367bb4) removes the ES6 class.